src/gui/core/canvas.cpp

```diff
@@ -272,7 +273,10 @@ void image_shape::draw(wfl::map_formula_callable& variables)
 	const std::string& name = image_name_(variables);
 
 	if(name.empty()) {
-		DBG_GUI_D << "Image: name is empty or contains invalid formula, will not be drawn.";
+		if(!failure_logged_) {
+			DBG_GUI_D << "Image: name is empty or contains invalid formula, will not be drawn.";
```

Comment:
Always annoying to see errors that tell you something is wrong but don't show the actual thing. Would be nice to log the formula as well.

Why is this on debug level when it sounds like an error anyway?

Reply:
I'm looking further at this, but will make it a separate PR. There's also documentation that needs checking to find out when it stopped being true; the current doc of `image_name_` says that "If it isn't a formula the image will be loaded in the constructor." but it's now always loaded in draw().

Reply:
The same log level is used for WFL parsing (yes, WFL parsing is in log-domain `gui/draw`), so if this log is seen there's already debugging directly above it.

There seems to be no access to the formula except via the logs, will ask on IRC if that's by design.